src/trait_solver/6.md

```diff
@@ -11,6 +11,29 @@
 {{#include ../../code/examples/stderr/trait_solver_6.stderr}}
 ```
 
-<!-- FIXME: explanation -->
+Let's first explicitly annotate all elided lifetimes:
+
+```rust
+fn func<'a>(a: &'a u32) {
+    dbg!(a);
+}
+
+fn accepts_func<'a>(b: for<'b> fn(&'b u32), c: &'a u32) {
+    b(c);
+}
+
+fn main() {
+    accepts_func(func as fn(_), &23);
+}
+```
+
+`func` is a higher-ranked function of type `for<'a> fn(&'a u32)`. This means it can accept *any* lifetime.
+`accepts_func`'s first parameter `b` is also a higher ranked function of the same type, so it needs a function that can accept any lifetime.
+
+But in `main`, we cast the higher-ranked function type to `fn(_)`. `fn(_)` is inferred to *not* be a higher-ranked function pointer type
+(inferring a *specific* lifetime for the argument), so it is no longer valid to pass to `accepts_func`.
```

Comment:
I'd probably say that `_` must be inferred to a concrete type. Because `fn` does not really affect inference in this case.